candle-transformers/src/models/whisper/audio.rs

```diff
@@ -102,22 +113,29 @@ fn log_mel_spectrogram_w<T: Float>(
     let half = T::from(0.5).unwrap();
     let mut fft_in = vec![zero; fft_size];
     let mut mel = vec![zero; n_len * n_mel];
+    let n_samples = samples.len();
+    let end = std::cmp::min(n_samples / fft_step + 1, n_len);
 
-    for i in (ith..n_len).step_by(n_threads) {
+    for i in (ith..end).step_by(n_threads) {
         let offset = i * fft_step;
 
         // apply Hanning window
-        for j in 0..fft_size {
-            fft_in[j] = if offset + j < samples.len() {
-                hann[j] * samples[offset + j]
-            } else {
-                zero
-            }
+        for j in 0..std::cmp::min(fft_size, n_samples - offset) {
+            fft_in[j] = hann[j] * samples[offset + j];
         }
 
-        // FFT -> mag^2
+        // fill the rest with zeros
+        if n_samples - offset < fft_size {
+            fft_in
+                .iter_mut()
+                .skip(n_samples - offset)
+                .for_each(|x| *x = zero);
```

Comment:
Couldn't we use the `fill` method on the slice `fft_in[n_samples-offset..]` for this rather than doing a `for_each`?

Reply:
oh yea thats a great point, thanks for the recommendation! latest commit uses fill